### DISCUSSION TITLE
Working Better Apart From A Few Things

### DISCUSSION THREAD

**Discussion by TEOL2023** [aclap-dev/video-downloadhelper#173] [Help and Q&A]
### Your question

This new version as well as the new converter is working a bit better now on my old Windows 10 i7 8GB RAM Quad Core banger.

However there are still a few buggy things.

192kbs HQ m4a Mixcloud Audio Downloads not available in any other software transform into 1GB Flac files under an exotic extension format which is easy to rename so I can get that far into listening to them but they start in winamp way into the show you download and it has in and out buffering from where it does start.

Spotify Web Player downloads the files with the full length and size but what you get is just a few seconds then it's blank for the rest of your 1-15 minutes.

Youtube and YouTube Music has a issue with only detecting the out of date 44100 audio formats at the moment as well. There is a github with a lot of codes on it to get it fixed though.

https://gist.github.com/AgentOak/34d47c65b1d28829bb17c24c04a0096f

Final issue I found was with Soundcloud it downloads fine but adds a little bit at the end where it can't be imported into audacity whilst my other snabbing method on the vi Soundcloud I can put it in my download manager it gets all the bytes and it allows me to import in audacity to turn the sound down if too high or up if too low just fine. 

Overall I am very happy with this version it gets a lot more of what everyone needs now when the previous version gave the proverbial up yours when WEBM was the highest detection. I would like to say thank you developers for ironing out those issues it's becoming more than just a software for getting hidden mp3s of on demand radio shows podcasts again but a few things I've tried does need fixing and you do need to use this github to fix the non detecting formats left as audios like 48000khz 160 ogg is major to get working and the github has the codes for how.

And there is quirks like the extra bit it puts at the end of soundclouds making it fail to be edited when vi soundcloud imports fine Mixcloud 192 failing to get the start and also having a Flac with exotic extension quirk and not being even to browse the content of the flac to attempt to rewind to the start and Spotify getting the start before flat noise.

But yeah, VDH is working fine for me again it's now getting most of the things it should get other than fancy formats of YouTube for stripping stuff out of it when you stop the sub and won't be able to watch your offline videos anymore unless they are already downloaded with VDH instead of in the browser or even the highest 48000khz 160 and 192k if it still exists that CYS format on basic non paid formats.

Audio only as 128k mp3 conversions is a definite no no for me I want to hit the audio option and get the real mc coy without the video conversions won't do for the modern times when all players play ogg and m4a which isn't watery tunnel sounding like 128 44100 mp3 is as a 128 in the updated codecs is double the res better in the sound most of the time and 256k is still MP3 HQ in the software when 320kbs is the true HQ on the old out of date codec.

I would like to say a big thanks for getting the software back to what it needs to be ok audio only isn't what I want at the moment and the converter still says 256 is HQ when it isn't anymore as since connections got faster 320 is the only HQ mp3 now and the audio only doesn't even give native but conversions and it isn't the 256k but a 128k that is below the 135k max vbr hit point on a vbr m4a and definitely behind the 160k 48000 version and below the 192 if it still exists as a non paid.

### Addon & CoApp version and details

Platform: Win 10 x86-64
Browser: Mozilla Firefox 122.0.1
CoApp: VdhCoApp 2.0.17
RAM: 8GB Quad
Processor: Intel i7

**Comment by paulrouget** [comment]
> 192kbs HQ m4a Mixcloud Audio Downloads not available in any other software transform into 1GB Flac files under an exotic extension format which is easy to rename so I can get that far into listening to them but they start in winamp way into the show you download and it has in and out buffering from where it does start.

Sorry, can you reformulate your paragraph? I have a hard time understanding what you mean.
What's the source of the stream? What's the exotic extension?

> Spotify Web Player downloads the files with the full length and size but what you get is just a few seconds then it's blank for the rest of your 1-15 minutes.

I'm not surprised. Did this use to work?

> Youtube and YouTube Music has a issue with only detecting the out of date 44100 audio formats at the moment as well. There is a github with a lot of codes on it to get it fixed though.
> https://gist.github.com/AgentOak/34d47c65b1d28829bb17c24c04a0096f

Can you try to allow more variants in the settings (More > Video Qualities)?

> Final issue I found was with Soundcloud it downloads fine but adds a little bit at the end where it can't be imported into audacity whilst my other snabbing method on the vi Soundcloud I can put it in my download manager it gets all the bytes and it allows me to import in audacity to turn the sound down if too high or up if too low just fine.

Can you explain a bit more. In soundcloud it adds extra bits at the end? What does audacity say?

> Overall I am very happy with this version it gets a lot more of what everyone needs now when the previous version gave the proverbial up yours when WEBM was the highest detection. I would like to say thank you developers for ironing out those issues it's becoming more than just a software for getting hidden mp3s of on demand radio shows podcasts again but a few things I've tried does need fixing and you do need to use this github to fix the non detecting formats left as audios like 48000khz 160 ogg is major to get working and the github has the codes for how.

Thank you :)

So that format thing, is youtube specific. I'm sure we can improve the situation. I'll need to figure that out.

> Audio only as 128k mp3 conversions is a definite no no for me I want to hit the audio option and get the real mc coy without the video conversions won't do for the modern times when all players play ogg and m4a which isn't watery tunnel sounding like 128 44100 mp3 is as a 128 in the updated codecs is double the res better in the sound most of the time and 256k is still MP3 HQ in the software when 320kbs is the true HQ on the old out of date codec.

Please be a little more specific. Where does that audio come from, and which conversions are you talking about?

> I would like to say a big thanks for getting the software back to what it needs to be ok audio only isn't what I want at the moment and the converter still says 256 is HQ when it isn't anymore as since connections got faster 320 is the only HQ mp3 now and the audio only doesn't even give native but conversions and it isn't the 256k but a 128k that is below the 135k max vbr hit point on a vbr m4a and definitely behind the 160k 48000 version and below the 192 if it still exists as a non paid.

Right, to be honest, i've been focusing on video for now.

I'm sure we can greatly improve the audio quality.

**Reply by TEOL2023** [reply]
Copy URL doesn't work anymore think thats a bug. But I can copy from details the 192kbs details from Mixcloud detection in VDH as well as the lower quality 64kbs to see if we can get some sort of proper working code for 192k HQ through hit details for now.

Mixcloud 192 HQ the VDH Hit Details the one to save as .oga and have to rename as .flac only to find it only plays from a bit in:

https://audio14.mixcloud.com/secure/dash_multi/,3/5/2/0/7351-3b26-46a8-a665-4f5834ae3b1c-64K,3/5/2/0/7351-3b26-46a8-a665-4f5834ae3b1c-192K,.m4a.urlset/manifest.mpd

Proper Details 64k from Mixcloud DL Site:

https://stream13.mixcloud.com/secure/c/m4a/64/c/4/8/4/7c33-afeb-4398-b9cf-b1aeeba7e1f0.m4a?sig=8awiITeKDVKorXGazXUE_A

As for Spotify I did used to use rtmpdumphelper rtmpsrv rtmpsuck back when the RTMPDumpHelper frontend for it used to work and it used to get stuff but VDH always struggled past those first seconds to be honest but gets the full length that used to and the intro of the actual sound though.

As for soundcloud vi soundcloud gives me a link like this and it doesn't put on any extra bytes which Audacity can't load in:

https://cf-media.sndcdn.com/TMIHysgGGEOS.128.mp3?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLW1lZGlhLnNuZGNkbi5jb20vVE1JSHlzZ0dHRU9TLjEyOC5tcDMqIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNzA4NTMzMjg5fX19XX0_&Signature=bnjLou13GJZJ2gDzhBNC3gX~tqxN0BBOScjQDw-Dm0az1hdK~7vmT7HRrKABNR4DPWOKdL6LbEe3V6zqRvdrnMFhpPW6QSJEo1sfzIhlkK-P4o4XDBGXUUGtEE081qEY394ZPSMq3pGniFZe0EX2MCxXKdF-69~I-q7BDDYfhzHgxvqn7w-PnHOSEvDtVlhs~6uSb6UPqT850~lbC9KvLf54~VdngTYeXJ6dTvoKacZm118qaFR7TCXx9ToyFo1Ap5Ej3ixNRkLQ6gs6OHyZfcYDPgyN2iagQ0ZKfHJDVypS7D8e9-2FdRhx6z3gcip9lUHLhsUfK9QESr-RelW1sg__&Key-Pair-Id=APKAI6TU7MMXM5DG6EPQ

The VDH lnk of the same that puts extra bytes on the end which means I can't turn the volume down when the mic is too loud on any podcast show:

https://cf-hls-media.sndcdn.com/playlist/TMIHysgGGEOS.128.mp3/playlist.m3u8?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLWhscy1tZWRpYS5zbmRjZG4uY29tL3BsYXlsaXN0L1RNSUh5c2dHR0VPUy4xMjgubXAzL3BsYXlsaXN0Lm0zdTgqIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNzA4NTMzNDc5fX19XX0_&Signature=fB51tgBmIMYEp4dH26YrH2svTaeBljNSLRyIqsSzxLcgm1rv1MVBYy7b7zWEx1GLmH8bFGAZCBmToH~Iieh-2LG4JiRgsrpJyWGWydi6mrBnB4KD2broeZCmdN16Tjti6jPxWM~l3QqPtGI0QTOl~ntGqIAEvWOsLTTN6N1aft~UongbCkX7UJe89rCVtYo-QW1OUjWtxJTzg7OBdMtF49vLouJzxAKgCURhMlfAM5JAKEn365YMgKBML2D50tPZz53odTsRwk4r3SDlPlDSzEOqnhHIq2HGTvhQ8pFyjYT6AGrXYZr09mHt~mXiXKa-3Qrs2wfGZ7eS0VuK-v-l7g__&Key-Pair-Id=APKAI6TU7MMXM5DG6EPQ&track_authorization=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJnZW8iOiJHQiIsInN1YiI6IjEyNTgwMzQyNSIsInJpZCI6IiIsImlhdCI6MTcwODUyMTMzMX0.-SFfYFXy-V2mqicyHZoE7O7dX5CNvommWRNty8E5HEM

As for the main point in this version of VDH the YouTube audio/video formats though as well as the audio only button giving me a conversion not the proper m4a or ogg whatevers better of the native formats instead giving me a 128/44100 mp3 conversion lower quality than a 135 max peek VBR m4a or 160 ogg that hadn't quite been uploaded in 160 to begin so gives us 128 48 on 160 but still that also has a higher maximum quality peak than 128 itself though clicking on that VDH audio button.

My in browser downloader gives links like this for native audio only formats that are not the mp3 conversions though if that helps on getting native formats on the audio buttons side working in VDH as that would be better for me than 128 44100 mp3 to not get the full detail quality of a 135 or higher VBR peak.

160 0gg Dash - 48000 (My IP Censored)

https://rr2---sn-cu-aigss.googlevideo.com/videoplayback?expire=1708543450&ei=evnVZZrUFfi-mLAPs8OHmAs&ip=10.000.000.000&id=o-AJwmpTQ-Mr1CMQ9-C2aWiLmO_vKnZD3t345mftI6gTv8&itag=251&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=vE&mm=31%2C29&mn=sn-cu-aigss%2Csn-cu-c9i6&ms=au%2Crdu&mv=m&mvi=2&pl=25&initcwndbps=1410000&siu=1&spc=UWF9fyfIwVryp6IRBHRHVA5tpLsD1M5R1U1Z2JlyXP4sleayzVn54r-PVP5uk2P9UXj7q8M&vprv=1&svpuc=1&mime=audio%2Fwebm&ns=yMSJlsYrZN3j1DBodFeieHIQ&gir=yes&clen=3466806&dur=217.981&lmt=1705266093844405&mt=1708521414&fvip=2&keepalive=yes&fexp=24007246&c=WEB&sefc=1&txp=5318224&n=W1Zble3h9jkyoZJ&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Csiu%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=APTiJQcwRQIgYHmNf3-mXSgkuu3vZ2mW2znst6Mujap3zXHPkiYN04gCIQCZC8_dv-Un8PlUXWQUy9RosSoeCsVjfTZTmYtW5jYkFA%3D%3D&ratebypass=yes&sig=AJfQdSswRQIgSOvo9HN5K5clRVpXbiLH2vYdq76k7ZDHn4FuHqIxr4sCIQD78fy2o8H-YIIEa16svcMfrWmGwf7clspXx_EF1O3f7g%3D%3D

Better m4a - 41000 (My IP Censored)

https://rr2---sn-cu-aigss.googlevideo.com/videoplayback?expire=1708543450&ei=evnVZZrUFfi-mLAPs8OHmAs&ip=10.000.000.000&id=o-AJwmpTQ-Mr1CMQ9-C2aWiLmO_vKnZD3t345mftI6gTv8&itag=140&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=vE&mm=31%2C29&mn=sn-cu-aigss%2Csn-cu-c9i6&ms=au%2Crdu&mv=m&mvi=2&pl=25&initcwndbps=1410000&siu=1&spc=UWF9fyfIwVryp6IRBHRHVA5tpLsD1M5R1U1Z2JlyXP4sleayzVn54r-PVP5uk2P9UXj7q8M&vprv=1&svpuc=1&mime=audio%2Fmp4&ns=yMSJlsYrZN3j1DBodFeieHIQ&gir=yes&clen=3528966&dur=218.012&lmt=1705266080955937&mt=1708521414&fvip=2&keepalive=yes&fexp=24007246&c=WEB&sefc=1&txp=5318224&n=W1Zble3h9jkyoZJ&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Csiu%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=APTiJQcwRQIgXXVh55MaxISTbTfOW5WSiPAW775t6Ktbn8lYHVPdtksCIQC-h8C5VjktaulnPXiaAesj6JGgzDhkjr0CHJFynGKHdg%3D%3D&ratebypass=yes&sig=AJfQdSswRQIhAIYZOIIULX_mfQijSabzBYpP4sapX32_7A5khArb42_qAiAwHZkvQaN1L1dCir8eznqh1sm5nh_0XIhk_ysJVKC8jg%3D%3D

1080 MP4 Video Only: This is only for a IP censored base code to work with the codes in the 2023-2024 format codes below.

https://rr2---sn-cu-aigss.googlevideo.com/videoplayback?expire=1708543450&ei=evnVZZrUFfi-mLAPs8OHmAs&ip=10.000.000.000&id=o-AJwmpTQ-Mr1CMQ9-C2aWiLmO_vKnZD3t345mftI6gTv8&itag=137&aitags=133%2C134%2C135%2C136%2C137%2C160%2C242%2C243%2C244%2C247%2C248%2C278&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=vE&mm=31%2C29&mn=sn-cu-aigss%2Csn-cu-c9i6&ms=au%2Crdu&mv=m&mvi=2&pl=25&initcwndbps=1410000&siu=1&spc=UWF9fyfIwVryp6IRBHRHVA5tpLsD1M5R1U1Z2JlyXP4sleayzVn54r-PVP5uk2P9UXj7q8M&vprv=1&svpuc=1&mime=video%2Fmp4&ns=yMSJlsYrZN3j1DBodFeieHIQ&gir=yes&clen=112424447&dur=217.960&lmt=1705265961865149&mt=1708521414&fvip=2&keepalive=yes&fexp=24007246&c=WEB&sefc=1&txp=5309224&n=W1Zble3h9jkyoZJ&sparams=expire%2Cei%2Cip%2Cid%2Caitags%2Csource%2Crequiressl%2Cxpc%2Csiu%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=APTiJQcwRQIhALF9DVCEoolVNgDXLKlWXxFGQTBL2zUBZiIaADYee2jmAiA0_qDguKSIs2WE4I623SXIn7Kx-F-0Sodd9I5PLITuIA%3D%3D&ratebypass=yes&sig=AJfQdSswRQIgNZTieQAaSvng2qW-69BYLpydmC2ZvnPsrhDufT1OV08CIQDh88B05hraB7966wN4JSV_nZQw7BaxXq-1R9paG9fJcw%3D%3D

1080 WEBM base code to use with webm codes:

https://rr2---sn-cu-aigss.googlevideo.com/videoplayback?expire=1708543450&ei=evnVZZrUFfi-mLAPs8OHmAs&ip=10.000.000.000&id=o-AJwmpTQ-Mr1CMQ9-C2aWiLmO_vKnZD3t345mftI6gTv8&itag=248&aitags=133%2C134%2C135%2C136%2C137%2C160%2C242%2C243%2C244%2C247%2C248%2C278&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=vE&mm=31%2C29&mn=sn-cu-aigss%2Csn-cu-c9i6&ms=au%2Crdu&mv=m&mvi=2&pl=25&initcwndbps=1410000&siu=1&spc=UWF9fyfIwVryp6IRBHRHVA5tpLsD1M5R1U1Z2JlyXP4sleayzVn54r-PVP5uk2P9UXj7q8M&vprv=1&svpuc=1&mime=video%2Fwebm&ns=yMSJlsYrZN3j1DBodFeieHIQ&gir=yes&clen=43927612&dur=217.960&lmt=1705265976491076&mt=1708521414&fvip=2&keepalive=yes&fexp=24007246&c=WEB&sefc=1&txp=530F224&n=W1Zble3h9jkyoZJ&sparams=expire%2Cei%2Cip%2Cid%2Caitags%2Csource%2Crequiressl%2Cxpc%2Csiu%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=APTiJQcwRQIhAKHNs3y9DO13ZLxXPLxWceImsEuMNWzKyrZKeAsEDWwSAiB6nl1lBNhmU3Lboo8a7_X_WUBgUoMeNB6MwjGfsSjXxw%3D%3D&ratebypass=yes&sig=AJfQdSswRAIgYByCwmrPOIZTkUgbBvk6dXPNgG-0T_mp6vlElubJgL4CIBw9oKArnZBSeWssmUwEp-1HHMuCebb6HpiMeu-DQJyZ

There is actually a new 2023-2024 version with the updated dash audio sound codes as well as video codes for YouTube around now that was out of date sorry about that.

But hopefully these updated dash audio codes and more importantly dash video as well would work in conjunction with the in browser app codes as that is out of date giving a slow loading and downloading experience whilst VDH is very snappy and good at what it can do once it gets all the formats in.

Making 10 the maximum of video qualities though also doesn't get us any of these bigger free formats and that probably means those on paid when they cancel their subs can't get VDH decrypted versions because even those probably don't pick up yet as the ones you get by increasing it to 10 isn't the best of the free formats yet.

https://gist.github.com/MartinEesmaa/2f4b261cb90a47e9c41ba115a011a4aa

I don't know how all these base codes from my URL copying alt browser software and site based grabbers for mix and sound clouds are going to help patching the code for the 192kbs as well as on the SC getting it directly without the extra bytes on the end but the github will definitely help patching existing VDH detection codes I'd imagine to get all those dash formats of both audio and video detecting on the YouTube. At the moment the audio seems to be watered down quality in the downloaded videos to what I'm getting from my audio only streams from my old clunky in browser software but even then that is too old to give me all the video formats it has in the list but is up to date enough to give me the 160 48 opus ogg sound at least to copy a base code for that dash audio.

Wow 35007k Dash Video the highest in the github list thats a big bitrate and 480 48000 Quadraphonic Opus. I doubt any video has both of them enabled but thats just some of the higher audio and visual combinations in dash currently we don't have detecting yet although I doubt anyone will be using that sort of insane quality on there anyway.

Ok, thats it for now but getting all these formats picking up would make VDH the most up to date in development YouTube Downloader and thats the idea in showing you some of the links my out of date browser one gives me as at least it's good enough that it's in date enough to get an audio only code for Opus 160/48 though as that seems to be the highest left standing in 2.0 stereo on the free users so the one that most of us gets.

I wonder if paid dash audio 256k m4a code 141 would work for all users if given a higher free to watch video quality with it as a download and combine combination in VDH I highly doubt it though. lol

**Comment by TEOL2023** [comment]
Just a report back the MixCloud 192 works from the beginning of a file but only with a workaround opening it in Audacity and converting it to any format 192.

As the downloaded files themselves is a weird ogg and flac hybrid which says ogg as a file and flac as a type all the normal softwares that can open flacs and oggs the files clash in and do that can't move around the file and starting a long way in and popping sound things.

Opening it in Audacity actually got the file as it's supposed to be so I can just save it as any format with a 192 bitrate and yeah Mixcloud HQ works fine then.

So now all we need is to is get that 48000 support on the YouTube we should have had that years ago the straight .WEBM extension HD one but better late than never.
